tasks/main.yml

```diff
@@ -129,3 +125,20 @@
     state: "{{ item.state | default(present) }}"
     reload: "{{ item.reload | default(False) }}"
   with_items: "{{ selinux_logins }}"
+
+- name: Get SELinux modules facts
+  selinux_modules_facts:
+
+- include_tasks: selinux_load_module.yml
+  vars:
+    name: "{{ item.name | default('') }}"
+    state: "{{ item.state | default('enabled') }}"
+    priority: "{{ item.priority | default('0') }}"
+    current_state: "{{ selinux_installed_modules[name][priority] | default('absent') }}"
+    store: "{{ selinux_policy | default(ansible_selinux.type, true) }}"
+  when: selinux_modules is defined
+  loop: "{{ selinux_modules | flatten(levels=1) }}"
+
+- name: Reload SELinux policy
+  command: semodule -R
+  when: ansible_selinux.status != "disabled"
```

Comment:
This should be done in a handler: https://docs.ansible.com/ansible/latest/user_guide/playbooks_handlers.html

create a `handlers/` directory in the root of your role
create a file `main.yml` in that directory
move this task to that file and add a `listen`:
```yaml

- name: Reload SELinux policy
  command: semodule -R
  listen: __selinux_reload_policy
  when: ansible_selinux.status != "disabled"
```
Then, in the tasks that modify the policy, add a `notify` like this:
```yaml
    - name: Install module with priority
      command: "semodule -n -s {{ store }} -X {{ priority }} -i {{ tempdir.path }}/{{ item.path | basename }}"
      when: priority != "0" and selinux_priorities
      notify: __selinux_reload_policy
```